sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -94,7 +94,10 @@ def get_historical_features(
             )
 
             table_id = _upload_entity_df_into_bigquery(
-                config.project, entity_df, client
+                config.project,
+                getattr(config.offline_store, "entity_dataset_name", "feast"),
```

Comment:
Are you using getattr in order to set a default? The configuration object should be providing the defaults instead of placing configuration in the code. Can we use a property method on the configuration object instead?

Reply:
This was necessary because right now you can have local provider defined in `feature_store.yaml` but still use BigQuery data source. Local provider will default to the file offline store config, which doesn't have `entity_dataset_name` property. This was a quick workaround that doesn't break any of our tests.

In a separate PR we can fix this by either allowing multiple offline store configs to be defined in the feature_store.yaml, or you can suggest something else. But I think we should merge this in first.

Reply:
After discussing this offline with Willem, we decided to enforce offline store during `feast apply`. This will restrict Feast repo to use a single offline store, and we can get rid of this ugly getattr.

While this is not an ideal solution, if a use case for multiple data sources appears in the future, we can think about removing this restriction then.